airbyte-workers/src/test/java/io/airbyte/workers/internal/EmptyAirbyteSourceTest.java

```diff
@@ -0,0 +1,399 @@
+/*
+ * Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.workers.internal;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.google.common.collect.Lists;
+import io.airbyte.commons.json.Jsons;
+import io.airbyte.config.ResetSourceConfiguration;
+import io.airbyte.config.State;
+import io.airbyte.config.WorkerSourceConfig;
+import io.airbyte.protocol.models.AirbyteGlobalState;
+import io.airbyte.protocol.models.AirbyteMessage;
+import io.airbyte.protocol.models.AirbyteMessage.Type;
+import io.airbyte.protocol.models.AirbyteStateMessage;
+import io.airbyte.protocol.models.AirbyteStateMessage.AirbyteStateType;
+import io.airbyte.protocol.models.AirbyteStream;
+import io.airbyte.protocol.models.AirbyteStreamState;
+import io.airbyte.protocol.models.ConfiguredAirbyteCatalog;
+import io.airbyte.protocol.models.ConfiguredAirbyteStream;
+import io.airbyte.protocol.models.StreamDescriptor;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class EmptyAirbyteSourceTest {
+
+  private EmptyAirbyteSource emptyAirbyteSource;
+  private final AirbyteMessage EMPTY_MESSAGE =
+      new AirbyteMessage().withType(Type.STATE)
+          .withState(new AirbyteStateMessage().withStateType(AirbyteStateType.LEGACY).withData(Jsons.emptyObject()));
+
+  private final ConfiguredAirbyteCatalog airbyteCatalog = new ConfiguredAirbyteCatalog()
+      .withStreams(Lists.newArrayList(
+          new ConfiguredAirbyteStream().withStream(new AirbyteStream().withName("a")),
+          new ConfiguredAirbyteStream().withStream(new AirbyteStream().withName("b")),
+          new ConfiguredAirbyteStream().withStream(new AirbyteStream().withName("c"))));
+
+  @BeforeEach
+  public void init() {
+    emptyAirbyteSource = new EmptyAirbyteSource();
+  }
+
+  @Test
+  public void testLegacy() throws Exception {
+    emptyAirbyteSource.start(new WorkerSourceConfig(), null);
```

Comment:
Can we also add a test where `WorkerSourceConfig.sourceConnectionConfiguration` is set to `Jsons.emptyObject()`? E.g. https://github.com/airbytehq/airbyte/blob/3bab4c9d1bebb763336d4819de341853c14e90e1/airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/activities/GenerateInputActivityImpl.java#L44